lib/ex_unit/test/ex_unit/case_test.exs

```diff
@@ -112,4 +114,29 @@ defmodule ExUnit.CaseTest do
       end
     end
   end
+
+  test "warns for using it twice with different options" do
+    assert capture_io(:stderr, fn ->
+        defmodule WarnsUsedTwice do
+          use ExUnit.Case
+          use ExUnit.Case, async: true
+        end
+      end) == ""
```

Comment:
Not sure if this is the expected behaviour, had to change the `__using__` macro a little bit to make this work.